plugins/modules/s3_object_info.py

```diff
@@ -722,16 +720,7 @@ def main():
     dualstack = module.params.get("dualstack")
 
     if dualstack and endpoint_url:
-        module.deprecate(
-            (
-                "Support for passing both the 'dualstack' and 'endpoint_url' parameters at the same "
-                "time has been deprecated."
-            ),
-            date="2024-12-01",
-            collection_name="amazon.aws",
-        )
-        if "amazonaws.com" not in endpoint_url:
-            module.fail_json(msg="dualstack only applies to AWS S3")
+        module.fail_json(msg="Parameters 'dualstack' and 'endpoint_url' are mutually exclusive.")
```

Comment:
```suggestion
```

Because we no longer need a deprecation warning, we can let AnsibleModule (AnsibleAWSModule) handle this.